customTests/azure_hw_topology_check.nhc

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+function load_expected_topology() {
+    TOPO_FILE="$1"
+
+    # Filter down to just numaids and <pci /> sections
+    filtered_topo=$(cat $TOPO_FILE | grep -o -E -e 'numaid=\"[[:digit:]]{1,}\"' -e '<pci.{1,}/>' )
+
+    # Set the initial value of numaid
+    numaid="0"
+
+    # Loop through each line of the input file
+    echo "$filtered_topo" | while read line; do
+        # Check if the line starts with "numaid"
+        if [[ $line == numaid=* ]]; then
+            # If it does, update the value of numaid
+            numaid="${line#numaid=}"
+        else
+            # If it doesn't, extract the busid and print the line with the current value of numaid
+            busid="$(echo $line | awk -F'"' '{print $2}')"
+            result="numaid=$numaid-busid=$busid"
+            result="${result//\"/}"
+            echo $result
+        fi
+    done
+}
+
+function get_actual_topology() {
+    raw_topo=$(lstopo)
```

Comment:
In some setups this will try and start lstopo graphically. Perhaps better to use "lstopo-no-graphics"

Reply:
Good call, I'll try this out

Reply:
Updated to use lstopo-no-graphics and also added in the relevant package install in custom-test-setup.sh